src/components/profileemailinfo.main.jsx

```diff
@@ -0,0 +1,180 @@
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import intl from 'react-intl-universal';
+import { login } from '../utils/AuthService';
+import { cortexFetch } from '../utils/Cortex';
+import './profileemailinfo.main.less';
+
+const Config = require('Config');
+
+class ProfileemailinfoMain extends React.Component {
+  static propTypes = {
+    profileInfo: PropTypes.objectOf(PropTypes.any).isRequired,
+    onChange: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const { profileInfo } = this.props;
+    const email = profileInfo._emails[0]._element ? profileInfo._emails[0]._element[0].email : '';
+    this.state = {
+      failedSubmit: false,
+      emailInEditMode: false,
+      email,
+    };
+    this.setEmail = this.setEmail.bind(this);
+    this.submitEmailChange = this.submitEmailChange.bind(this);
+  }
+
+  setEmail(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  cancelEmail() {
+    this.setState({ emailInEditMode: false });
+  }
+
+  editEmail() {
+    this.setState({
+      emailInEditMode: true,
+    });
+  }
+
+  submitEmailChange(event) {
+    event.preventDefault();
+    const { email } = this.state;
+    const { profileInfo } = this.props;
+    login()
+      .then(() => {
+        cortexFetch(profileInfo._emails[0]._emailform[0].self.uri, {
+          method: 'post',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+          body: JSON.stringify({ email }),
+        })
+          .then((response) => {
+            if (response.status === 400) {
+              this.setState({ failedSubmit: true });
+            } else if (response.status === 201 || response.status === 200 || response.status === 204) {
+              this.cancelEmail();
+              const { onChange } = this.props;
+              onChange();
+            }
+            return response;
+          })
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error.message);
+          });
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(error.message);
+      });
+  }
+
+  render() {
+    const { emailInEditMode, failedSubmit } = this.state;
+    const { profileInfo } = this.props;
+    const email = profileInfo._emails[0]._element ? profileInfo._emails[0]._element[0].email : '';
```

Comment:
`profileInfo ` can be undefined at this point. Need to do a proper check of `profileInfo` as well.